spark-cassandra-connector/src/main/scala/org/apache/spark/metrics/CassandraConnectorSource.scala

```diff
@@ -20,8 +20,25 @@ object CassandraConnectorSource extends Source {
 
   val readByteMeter = metricRegistry.meter("read-byte-meter")
   val readRowMeter = metricRegistry.meter("read-row-meter")
-  val readPageWaitTimer = metricRegistry.timer("read-page-wait-timer")
   val readTaskTimer = metricRegistry.timer("read-task-timer")
 
-  lazy val ensureInitialized = SparkEnv.get.metricsSystem.registerSource(this)
+  CassandraConnectorSource._instance = Some(this)
+  CassandraConnectorSource._env = SparkEnv.get
```

Comment:
This is something potentially very dangerous. There can be multiple `CassandraConnectorSource` instances fighting for the shared "_instance" mutable field, overwriting it on their own instantiation. I believe you assume there is going to be only one, but the code doesn't force it in any way.